src/Behat/Testwork/Call/RuntimeCallee.php

```diff
@@ -42,21 +42,13 @@ class RuntimeCallee implements Callee
     /**
      * Initializes callee.
      *
-     * @param callable    $callable
+     * @param callable $callable
      * @param null|string $description
      *
      * @throws BadCallbackException If invalid callback provided
      */
     public function __construct($callable, $description = null)
     {
-        if (!is_array($callable) && !is_callable($callable)) {
```

Comment:
If it is a `callable` then by definition it cannot be anything other than these two options, so this check is redundant. We should not be trying to check the types of arguments at run time, this should be done by static analysis tools or tests

Reply:
The proper fix is to use a `callable` parameter type (which was not possible due to the PHP requirement in older Behat versions).

Reply:
Needs to be `callable|array` because the arrays that we receive are not the new modern callables (object, string) but instead the old callables (string, string) and php throws an error if you just use `callable` for this case

Reply:
This is not true. `callable` is really any callable.

Unless we support passing a class name instead of an object for the first array item of callables meant to call a non-static method on an instance that will be created later. But then it means that the phpdoc is wrong.

Reply:
> Unless we support passing a class name instead of an object for the first array item of callables meant to call a non-static method on an instance 

Yes, that is exactly what I meant, sorry if I did not explain it correctly.

Reply:
Maybe we should deprecate calling with an old-style `[string,string]`?

Reply:
I think that before deprecating it we should probably change our code to use [object, string] arrays as currently we use [string, string] everywhere

Reply:
Ah, I hadn't realised we did - yes, definitely.